domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs

```diff
@@ -60,8 +61,11 @@ internal class ContractServiceImpl : IContractService
                 return new GsrnNotFound();
             }
 
-            var overlappingContract = contractsByGsrn.FirstOrDefault(c =>
+            var contractsGsrn = contractsByAllGsrn.Where(c => c.GSRN == contract.GSRN).ToList();
```

Comment:
What happens if the request contains multiple contracts on the same GSRN number? 

It looks like contractsByAllGsrn variable is not updated with contracts created as part of the request handling?

Reply:
fixed in 1eb3207b

Reply:
I don't see a test where multiple contracts with same GSRN are bulk-created :)